src/Controller/UserController.php

```diff
@@ -30,47 +34,81 @@ public function __construct(RendererInterface $renderer, UserService $questionIn
         $this->userService = $questionInstanceService;
     }
 
-    public function showLogin()
+    /**
+     * @return Response
+     */
+    public function showLogin(): Response
     {
         return $this->renderer->renderView("login.html", []);
     }
 
-    public function showUsers(Request $request, array $requestAttributes)
+    /**
+     * This method returns a Response with the users searched with the specified page and text
+     *
+     * @param Request $request
+     * @param array $requestAttributes
+     * @return Response
+     */
+    public function showUsers(Request $request, array $requestAttributes): Response
     {
-        $arguments['currentPage'] = (int)$request->getParameter('page');
-        $arguments['pages'] = $this->userService->getUserNumber($requestAttributes);
+        $email = $this->userService->getFromParameter('email', $request, "");
+        $role = $this->userService->getFromParameter('role', $request, "");
+        $currentPage = (int)$this->userService->getFromParameter('page', $request, 1);
+        $totalResults = (int)$this->userService->getEntityNumberOfPagesByField(User::class, ['email' => $email, 'role' => $role]);
+        $resultsPerPage = 5;
+
+        $paginator = new Paginator($totalResults, $currentPage, $resultsPerPage);
+        $paginator->setTotalPages($totalResults, $resultsPerPage);
+
+        $arguments['email'] = $email;
```

Comment:
you are building this array just to send it to the render method.you could always:
```
return $this->renderer->renderView("admin-users-listing.phtml", [
  'email' => $email
]);

Reply:
$arguments array holds all the data(email, username, dropdownRole, paginator and users) not just email.
Are you suggesting something like this? 

`return $this->renderer->renderView("admin-users-listing.phtml", [
'email' => $email,
'usarname'=>$this->userService->getName(),
'dropdownRole' => $role,
'paginator' => $paginator,
'users' => $this->userService->getEntitiesByField(User::class, ['email' => $email, 'role' => $role], $currentPage, resultsPerPage)
]);`

Reply:
with proper formatting, yes
```
return $this->renderer->renderView("admin-users-listing.phtml", [ 
    'email' => $email, 
    'usarname'=>$this->userService->getName(), 
    'dropdownRole' => $role, 
    'paginator' => $paginator, 
    'users' => $users,
]);
```